### PR TITLE
Introduce EJBCA UpstreamAuthority plugin for SPIRE Server

### PR DESCRIPTION
**Pull request by m8rmclaren**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

This PR introduces [EJBCA](https://www.ejbca.org/) as a Server UpstreamAuthority plugin. This plugin uses a connected EJBCA to issue intermediate certificates for the SPIRE server.

> Per #4163, the EJBCA UpstreamAuthority plugin is compatible with [EJBCA Community](https://www.ejbca.org/).

**Description of change**
* Adds EJBCA as a built-in UpstreamAuthority plugin.
* Adds the [EJBCA Go Client SDK](https://github.com/Keyfactor/ejbca-go-client-sdk/tree/main) as a required dependency for communicating with the EJBCA REST API.
* Updates the `golang.org/x/oauth2` to `v0.21.0` required when importing the EJBCA Go Client SDK.

**Which issue this PR fixes**

### CODE REVIEW

File: pkg/server/plugin/upstreamauthority/ejbca/ejbca.go [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+	return catalog.MakeBuiltIn(pluginName,
+		upstreamauthorityv1.UpstreamAuthorityPluginServer(p),
+		configv1.ConfigServiceServer(p),
+	)
+}
+
+// Config defines the configuration for the plugin.
+type Config struct {
+	Hostname               string          `hcl:"hostname" json:"hostname"`
+	CaCert                 string          `hcl:"ca_cert" json:"ca_cert"`
```
[the remark this answers is not in the document]
Comment (m8rmclaren) [reply pkg/server/plugin/upstreamauthority/ejbca/ejbca.go:77]: It's an excellent point - I will remove the option to specify the `ca_cert` in the server config file.

File: pkg/server/plugin/upstreamauthority/ejbca/ejbca.go [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+	EndEntityProfileName   string          `hcl:"end_entity_profile_name" json:"end_entity_profile_name"`
+	CertificateProfileName string          `hcl:"certificate_profile_name" json:"certificate_profile_name"`
+	DefaultEndEntityName   string          `hcl:"end_entity_name" json:"end_entity_name"`
+	AccountBindingID       string          `hcl:"account_binding_id" json:"account_binding_id"`
+}
+
+type CertAuthConfig struct {
+	ClientCert     string `hcl:"client_cert" json:"client_cert"`
+	ClientCertPath string `hcl:"client_cert_path" json:"client_cert_path"`
+	ClientKey      string `hcl:"client_key" json:"client_key"`
```
[the remark this answers is not in the document]
Comment (m8rmclaren) [reply pkg/server/plugin/upstreamauthority/ejbca/ejbca.go:91]: Same as `ca_cert` - I will also remove `client_cert` and `client_key` as configuration values in the server config file. Thank you for pointing this out.

**Comment by m8rmclaren** [comment]
Hey everyone, here's the result of the EJBCA UpstreamAuthority integration test running on my Mac:
image

**Comment by m8rmclaren** [comment]
Hi @amartinezfayo -

I made the recommended changes to `ejbca.Config` and resolved what seemed like the issue in the EJBCA integration test. Same as before, the test succeeds on my Mac.

**Comment by amartinezfayo** [comment]
Hi @m8rmclaren, just checking in here. Do you need any help or have questions about the latests comments?

**Comment by m8rmclaren** [comment]
Hi @amartinezfayo!

I believe I've resolved the suggestions from your review, with the most important change being the removal of OAuth as an auth method. I also resolved the issue in the EJBCA integration test, and verified that the [GitHub Actions run in my fork](https://github.com/m8rmclaren/spire/actions/runs/10364287224).

**Pull request closed without merging**